### DISCUSSION TITLE
Email SMTP

### DISCUSSION THREAD

**Discussion by RachelBrodsky** [inducer/relate#868] [Q&A]
We're having trouble getting the email smtp to work. 

We added the gmail smtp server and connected it to an email account. This worked fine for local deployment, but when we tested the email sign in functionality with the publicly deployed application, the no-reply email sent a link to the localhost version of the application. How do we fix the email so it links users to the publicly deployed Relate?

<img width="521" alt="Screen Shot 2021-12-28 at 3 30 46 PM" src="https://user-images.githubusercontent.com/67277602/147604830-33e0e974-9117-45a1-a091-da5025ed4c58.png">

**Comment by inducer** [comment]
Relate doesn't know its public address from configuration, it infers it from the HTTP requests it receives. In your case, it looks like you might be using a reverse proxy of some sort, and so you need to configure the reverse proxy to pass on the relevant information to the app server. Nginx as a reverse proxy (see [documentation](https://documen.tician.de/relate/misc.html#configuring-nginx)) does the right thing on its own. See the [`build_absolute_uri`](https://docs.djangoproject.com/en/4.0/ref/request-response/#django.http.HttpRequest.build_absolute_uri) method in Django for details.

**Comment by dzhuang** [comment]
There's a setting item named `RELATE_BASE_URL`, make sure you've configured it in your `local_settings.py` file like this:

    RELATE_BASE_URL = "https://my.website.com"
